### PR TITLE
Serve SPA in AWS S3 and Cloudfront Services

### PR DESCRIPTION
**Pull request by ElizavetaRazumenko**
## What is done:

- Add and configure S3 bucket creation and website deployment using AWS CDK
- Add and configure CloudFront Distribution and Invalidation using AWS CDK
- update README.md

[Link to S3 bucket website](http://liza-aws-react-web-app-bucket.s3-website.eu-central-1.amazonaws.com/)

[Link to CloudFront website](https://dvtolf4nt67kd.cloudfront.net)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,3 +36,11 @@ Runs tests in console, in browser or with coverage.
...
 
 Runs linting and formatting for all files in `src` folder.
+
+### `npm run deploy`
+
+Add and configure S3 bucket creation and website deployment using AWS CDK and CloudFront Distribution using AWS CDK
+
+### `npm run destroy`
+
+CloudFront Invalidation using AWS CDK
```
Comment (furiousme) [inline README.md]: This command should destroy the stack, not invalidate paths

File: invalidate-cache.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const { execSync } = require('child_process');
+const stackName = 'AwsSdkStack';
+const outputKey = 'DistributionId';
+
+const distributionId = execSync(`aws cloudformation describe-stacks --stack-name ${stackName} --query "Stacks[0].Outputs[?OutputKey=='${outputKey}'].OutputValue" --output text`).toString().trim();
+
+execSync(`aws cloudfront create-invalidation --distribution-id ${distributionId} --paths '/*'`, { stdio: 'inherit' });
```
Comment (furiousme) [inline invalidate-cache.ts]: Could you please explain why this code is needed? I guess the following lines should make invalidations as well, or maybe I just don't get it

```
  new s3Deployment.BucketDeployment(this, 'DeployWebsite', {
        sources: [s3Deployment.Source.asset('../dist')],
        destinationBucket: bucket,
        distribution,
        distributionPaths: ['/*'],
      });
 ```

**Review by furiousme — COMMENTED** [review]
The behavior of the links you provided is very strange. First, both of them returned errors. Now both of them show the app. 

Screenshot 2024-06-10 at 15 07 11

Screenshot 2024-06-10 at 15 07 21

File: README.md (hunk 1 above)
Comment (ElizavetaRazumenko) [reply README.md]: I updated the code according to your comments

File: invalidate-cache.ts (hunk 2 above)
Comment (ElizavetaRazumenko) [reply invalidate-cache.ts]: I updated the code according to your comments

File: aws_sdk/lib/aws_sdk-stack.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      });
+  
+      new s3Deployment.BucketDeployment(this, 'DeployWebsite', {
+        sources: [s3Deployment.Source.asset('../dist')],
+        destinationBucket: bucket,
+        distribution,
+        distributionPaths: ['/*'],
+      });
+  
+      new cdk.CfnOutput(this, 'BucketURL', {
```
Comment (daespuor) [inline aws_sdk/lib/aws_sdk-stack.ts:35]: Why is this required?

File: aws_sdk/lib/aws_sdk-stack.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import * as cloudfrontOrigins from 'aws-cdk-lib/aws-cloudfront-origins';
+
+export class AwsSdkStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+       const bucket = new s3.Bucket(this, 'AwsReactWebAppBucket', {
+        bucketName: 'liza-aws-react-web-app-bucket',
+        websiteIndexDocument: 'index.html',
+        publicReadAccess: true,
```
Comment (daespuor) [inline aws_sdk/lib/aws_sdk-stack.ts:15]: This is not really public, right?

**Review by daespuor — COMMENTED** [review]
I left some questions. Also, the cloudfront site is not loading, could you verify please?
Screenshot 2024-06-13 at 11 51 55 AM
